### PR TITLE
Adding a dev container and getting-started-with-codespaces document

### PR DESCRIPTION
**Pull request by samruddhikhandale**
Hi 👋 

I am Samruddhi, a Software Engineer for GitHub, working with the [GitHub Codespaces](https://github.com/features/codespaces) team, and a maintainer for the [dev containers](https://github.com/devcontainers) org 👩‍💻

I am following up from a [Twitter thread](https://twitter.com/LostInTangent/status/1612207073760280578) with Jonathan Carter and contributing a dev container to this repo.

Thanks,
Samruddhi

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    },
+
+    // Uncomment to install NVIDIA CUDA - required for a GPU-powered codespace
+    // "features": {
+    //     "ghcr.io/devcontainers/features/nvidia-cuda:1": {
+    //         "installCudnn": true
+    //     }
+    // },
+
+    "hostRequirements": {
```
Comment (joshspicer) [inline .devcontainer/devcontainer.json:15]: There's a way to require a GPU sku, right? Should we have that commented out here too for folks that would want to level up to a GPU codespace?

File: .devcontainer/welcome-message.txt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+👋 Welcome to "Practical Deep Learning for Coders" in Codespaces!
+
+🛠️  Your environment is fully setup with all the required softwares and machine learning libraries.
```
Comment (joshspicer) [inline .devcontainer/welcome-message.txt:3]:
```suggestion
🛠️  Your environment is fully setup with all the required software and machine learning libraries.
```

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (samruddhikhandale) [reply .devcontainer/devcontainer.json:15]: I have already added the details to the getting-started-with-codespaces.md doc.
Linked in here as well. thanks!

**Comment by samruddhikhandale** [comment]
Hi @jph00 👋 

Wondering if you have some time to provide feedback on this PR, I'd appreciate that.
Thank you!

File: getting-started-with-codespaces.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
+# Using Codespaces to work with the "Practical Deep Learning for Coders" course
+
+
+To get started, create a codespace for this repository by clicking this 👇 
+
+[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://github.com/codespaces/new?hide_repo_select=true&ref=master&repo=485606685)
+
+A codespace will open in a web-based version of Visual Studio Code.
+
+**Note**: Dev containers is now an open spec which is supported by [GitHub Codespaces](https://github.com/codespaces) and [other supporting tools](https://containers.dev/supporting).
```
Comment (jph00) [inline getting-started-with-codespaces.md:10]:
```suggestion
**Note**: Dev containers is an open spec which is supported by [GitHub Codespaces](https://github.com/codespaces) and [other supporting tools](https://containers.dev/supporting).
```

File: getting-started-with-codespaces.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+## Opening a notebook
+
+The [dev container](.devcontainer/devcontainer.json) is fully configured with softwares and [machine learning libraries](.devcontainer/requirements.txt) needed for this course.
```
Comment (jph00) [inline getting-started-with-codespaces.md:14]:
```suggestion
The [dev container](.devcontainer/devcontainer.json) is fully configured with software and [machine learning libraries](.devcontainer/requirements.txt) needed for this course.
```

**Comment by jph00** [comment]
Many thanks @samruddhikhandale . Other than a couple of minor changes above, my only requests are to remove the mention of Codespaces from all notebooks except the 01 intro notebook - that's because all the other notebooks require GPU, and we only specifically call out services with free GPU options. My understanding is that Codespaces doesn't currently have a free GPU option (and furthermore to use the paid option would require manually editing the configuration to make it work). Since our focus is on making deep learning accessible, we try to ensure all our recommended approaches are freely available.

**Comment by samruddhikhandale** [comment]
Thanks @jph00, appreciate your feedbacks! 

> My understanding is that Codespaces doesn't currently have a free GPU option

Currently, GPU-powered Codespaces are free while it's in beta. One can get access by filling up a sign form as mentioned [in this doc](https://github.com/fastai/course22/pull/46/files#diff-c4da5c7d48846bdc3321eb25890fa014d620e7c610f40235dd11ab63491846c5R22).

ℹ️ I understand that GPU is required for majority of the notebooks and it can highly impact performance, however I was able to run all the notebooks except 09 and 10 without GPU access.

> and furthermore to use the paid option would require manually editing the configuration to make it work

I understand that it could be a friction to edit the config, however, I can [uncomment the configuration](https://github.com/fastai/course22/pull/46/files#diff-24ad71c8613ddcf6fd23818cb3bb477a1fb6d83af4550b0bad43099813088686R8-R13) so that it works with GPU and CPU codespaces. As adding the `nvidia-cuda` Feature could slow down Codespaces creations by two minutes, I wasn't sure how many people would enable GPUs as it is not generally available. Hence, I chose codespace creation performance over the need to update the config. 

Let me know what you think, I can go ahead and remove the references and make any additional changes if needed.

**Comment by jph00** [comment]
As you mention, it's not generally available -- but if there's a workable free tier after it's generally available, I'd be delighted to feature codespaces alongside kaggle, colab, and sagemaker studio wherever they're mentioned.

Whilst I understand it's technically possible to run without GPU, it's not the quality of experience I want our students to have.

Therefore i'd prefer to remove those mentions.

**Comment by samruddhikhandale** [comment]
Yep, that makes sense to me. Removed the references, I can follow up once GPUs are generally available, thanks!

**Comment by jph00** [comment]
Thank you for your contribution :)

**Pull request merged**